Test/Utils/ContrailAPI/ConfigureDNS.ps1

```diff
@@ -0,0 +1,174 @@
+. $PSScriptRoot\Constants.ps1
+. $PSScriptRoot\Global.ps1
+. $PSScriptRoot\VirtualDNSData.ps1
+. $PSScriptRoot\VirtualDNSRecordData.ps1
+
+function Set-IpamDNSMode {
+    Param (
+        [Parameter(Mandatory = $true)] [string] $ContrailUrl,
+        [Parameter(Mandatory = $true)] [string] $AuthToken,
+        [Parameter(Mandatory = $true)] [string[]] $IpamFQName,
+        [Parameter(Mandatory = $true)]
+            [ValidateSet("none","default-dns-server","tenant-dns-server","virtual-dns-server")]
+            [string] $DNSMode,
+        [Parameter(Mandatory = $false)] [string[]] $TenantServersIPAddresses = @(),
+        [Parameter(Mandatory = $false)] [string] $VirtualServerName
+    )
+
+    $IpamUuid = FQNameToUuid -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                    -Type 'network-ipam' -FQName $IpamFQName
+
+    $Request = @{
+        "network-ipam" = @{
+            "network_ipam_mgmt" = @{"ipam_dns_method" = $DNSMode}
+            "virtual_DNS_refs"  = @()
+        }
+    }
+
+    if ($DNSMode -ceq 'tenant-dns-server')     {
+        $DNSServer = @{
+            "ipam_dns_server" = @{
+                "tenant_dns_server_address" = @{
+                    "ip_address" = $TenantServersIPAddresses
+                }
+            }
+        }
+
+        $Request."network-ipam"."network_ipam_mgmt" += $DNSServer
+    }
+    elseif ($DNSMode -ceq 'virtual-dns-server')     {
+        if(-not $VirtualServerName) {
+            throw "You need to specify Virtual DNS Server to be used"
+        }
+
+        $VirtualServerFQName = @("default-domain", $VirtualServerName)
+
+        $DNSServer = @{
+            "ipam_dns_server" = @{
+                "virtual_dns_server_name" = [string]::Join(":",$VirtualServerFQName)
+            }
+        }
+        $Request."network-ipam"."network_ipam_mgmt" += $DNSServer
+
+        $DNSServerUuid = FQNameToUuid -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+                            -Type 'virtual-DNS' -FQName $VirtualServerFQName
+        $DNSServerUrl = $ContrailUrl + "/virtual-DNS/" + $DNSServerUuid
+        $DNSServerRef = @{
+            "href" = $DNSServerUrl
+            "uuid" = $DNSServerUuid
+            "to"   = $VirtualServerFQName
+
+        }
+        $Request."network-ipam"."virtual_DNS_refs" = @($DNSServerRef)
+    }
+
+    $RequestUrl = $ContrailUrl + "/network-ipam/" + $IpamUuid
+    $Response = Invoke-RestMethod -Uri $RequestUrl -Headers @{"X-Auth-Token" = $AuthToken} `
+        -Method Put -ContentType "application/json" -Body (ConvertTo-Json -Depth $CONVERT_TO_JSON_MAX_DEPTH $Request)
+
+    return $Response.'network-ipam'.'uuid'
+}
```

Comment:
This function is pretty long, can we split it to smaller functions? Especially, since we modify state of a single struct throughout the whole body. It makes it hard to track.

Reply:
Maybe something like:
```
Set-IpamDNSMode {
   $Request = {...}
   if (tenant) { $Request =  AddTenantStuff($Request) }
   elif (virtual) { $Request = AddVirtualStuff($Request) }
   ...
}
```